pmd-core/src/main/java/net/sourceforge/pmd/internal/Slf4jSimpleConfiguration.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.internal;
+
+import java.lang.reflect.Method;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+import org.slf4j.LoggerFactoryFriend;
+import org.slf4j.event.Level;
+
+public final class Slf4jSimpleConfiguration {
+    private Slf4jSimpleConfiguration() { }
+
+    public static void reconfigureDefaultLogLevel(Level level) {
+        System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", level.toString());
+
+        // Call SimpleLogger.init() by reflection.
+        // Alternatively: move the CLI related classes into an own module, add
+        // slf4j-simple as a compile dependency and create a PmdSlf4jSimpleFriend class in
+        // the package org.slf4j.simple to gain access to this package-private init method.
+        ILoggerFactory loggerFactory = LoggerFactory.getILoggerFactory();
+        Class<? extends ILoggerFactory> loggerFactoryClass = loggerFactory.getClass();
+        try {
+            Class<?> simpleLoggerClass = loggerFactoryClass.getClassLoader().loadClass("org.slf4j.simple.SimpleLogger");
+            Method initMethod = simpleLoggerClass.getDeclaredMethod("init");
+            initMethod.setAccessible(true);
+            initMethod.invoke(null);
+        } catch (ReflectiveOperationException ex) {
+            System.err.println("Error while initializing logging: " + ex);
```

Comment:
I'm not sure with the reflection logic... doesn't this couple us a particular implementation? Is slf4j-impl always provided by every implementation?

Reply:
That's a good question. So, when the user runs "our" distribution, we are in control. Currently it is slf4j-simple, this is brought in via a dependency in pmd-dist and bundled into the zip binary distribution. This method here is called only when PMD is run from command line (PMD.runPMD) or via ant. It's not called, if PMD is run via PMD.processFiles.
But if someone would create an own distribution and replace the logger impl, then it would probably fail here - so I guess we should check, which logger impl we have and if it is not slf4j-simple, simply do nothing (then the --debug flag won't work).

Reply:
I've added a simple check